server/Arcadia.Assistant.Calendar.Notifications/DictionaryExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace Arcadia.Assistant.Calendar.Notifications
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public static class DictionaryExtensions
+    {
+        public static IDictionary<T1, T2> Merge<T1, T2>(
```

Comment:
wtf? `target.Concat(source).GroupBy(x => x.key).ToDictionary(d => d.Key, d => d.Last().Value);`

Reply:
you realize extension methods are evil? `class DictionaryMerge(IDictionary a, IDictionary b).Perfrom()`

Reply:
Why are the evil? If I know that this method can be applied to any dictionaries and arrays, why not?

Reply:
they cannot be DI'ed (you probably don't need to anyway :) ).
Merge here is not the worst example though. Your parser is.
You don't write static class with static methods even if you know that they won't contain any static fields, right?

Reply:
the proper way to use extensions is fluent interfaces :)

Reply:
Also, all dictionary methods change the object itself. this one does not and it might confuse.

Reply:
But what do you have against
> static class with static methods even if you know that they won't contain any static fields

?

Reply:
Done.